src/widgets/table/table.rs

```diff
@@ -1484,7 +1364,264 @@ mod tests {
                 .rows(vec![])
                 .footer(Row::new(vec!["h", "i"]))
                 .column_spacing(0);
-            assert_eq!(table.get_columns_widths(10, 0), &[(0, 5), (5, 5)])
+            assert_eq!(table.get_columns_widths(10, 0), [(0, 5), (5, 5)])
+        }
+
+        #[test]
+        fn spacers_priority_unconstrained() {
+            fn test(
+                highlight_spacing: HighlightSpacing,
+                selection: Option<usize>,
+                expected: Buffer,
+            ) {
+                let table = Table::default()
+                    .rows(vec![Row::new(vec!["hello", "world"])])
+                    .highlight_spacing(highlight_spacing)
+                    .highlight_symbol(">>>")
+                    .column_spacing(0);
+                let area = Rect::new(0, 0, 15, 3);
+                let mut buf = Buffer::empty(area);
+                if let Some(s) = selection {
+                    let mut state = TableState::default().with_selected(Some(s));
+                    StatefulWidget::render(table, area, &mut buf, &mut state);
+                    assert_buffer_eq!(buf, expected);
+                } else {
+                    Widget::render(table, area, &mut buf);
+                    assert_buffer_eq!(buf, expected);
+                }
+            }
```

Comment:
Continuing, here there's a chunk of code which provides context for each thing tested below. The reader has to memorize how this works and then apply that to the tests below in order to understand what's being tested.

This would be better pulled out as a function that just does the rendering.

Given that there are two paths, and the path chosen is based on passing either `None` or `Some(_)` those paths would be better expressed with two obvious methods - `render()`, `render_with_selection()`

Reply:
Done

Reply:
Thanks for this suggestion. I think we should add examples to the website for how to write tests when submitting a PR.